SEL32/sel32_defs.h

```diff
@@ -21,9 +21,99 @@
    CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 */
+/********************Attention************************/
+/* define the environment wanted for sel32 execution */
+/*                                                   */
+/* define CPUONLY to only run with CPU model         */
+/* undef CPUONLY to run with an IPU (2nd CPU)        */
+/*                                                   */
+/* for IPU models, define the IPU support method     */
+/* define USE_IPU_THREAD to run IPU in a thread      */
+/* undef USE_IPU_THREAD to run IPU in forked task    */
+/*                                                   */
+/* for threaded IPU models, select thread locking    */
+/* define USE_POSIX_SEM for POSIX semaphores         */
+/* undef USE_POSIX_SEM to use pthread mutexs         */
+/********************Attention************************/
+
+/* define CPUONLY to run without IPU */
+//#define CPUONLY                         /* run on CPU only */
+#undef CPUONLY                          /* run with cpu/ipu on system */
+
+/* undefine CPUONLY to run with IPU */
+/* define USE_IPU_THREAD to use IPU thread code instead of fork code */
+/* define USE_POSIX_SEM for POSIX semaphores otherwise use pthread mutex */
+/* forked mode IPU must only use semaphores */
+#ifndef CPUONLY
+//#define USE_POSIX_SEM                   /* use POSIX semaphores, else pthread mutex */
+//#undef  USE_IPU_THREAD                  /* run IPU as a forked sel32 */
+#define USE_IPU_THREAD                  /* run IPU in sel32_ipu.c thread */
+#endif
+
+/* enforce proper option combinations for CPU/IPU */
+#ifndef CPUONLY
+#define DEFINE_IPU_MODELS               /* IPU devices must be define for IPU */
+#ifndef USE_IPU_THREAD
+#define USE_POSIX_SEM                   /* forked mode IPU can only use semaphores */
+#endif
+#else /* CPUONLY */
+#undef  USE_IPU_THREAD                  /* make sure IPU code undefined for CPUONLY */
+#undef DEFINE_IPU_MODELS                /* make sure IPU models undefine too */
+#endif /* CPU_ONLY */
+
+/* use correct variable type for thread IPU */
+#ifdef USE_IPU_THREAD
+#ifdef USE_IPU_CODE
+#define LOCAL static                    /* IPU in thread needs static variables */

```

Comment:
I don't understand this.

"static" is about scope.  It has nothing to do with threading.  A variable defined at file scope has a single instance for the whole process, whether "static" or not.  There is such a thing as "thread local storage" (at least with some compilers and some platforms) but that isn't done with "static".

Reply:
Bad choice for a comment.  All of the CPU/IPU registers are the same name.  Making the IPU names static allows the same code to reference different variables.  Not thread specific, but needed for IPU code.  I have lots of other changes I am making so all of these items can be addressed with the next push.  Lets merge these changes and move on.

Reply:
Did you mean lots of changes on this topic, or more changes for other purposes?  In other words, is what we have here a complete feature/fix delivery?